src/DependencyInjection/ConnecthollandUserExtension.php

```diff
@@ -25,4 +29,91 @@ public function load(array $configs, ContainerBuilder $container)
 
         $loader->load('services.yaml');
     }
+
+    public function prepend(ContainerBuilder $container)
+    {
+        $bundles = $container->getParameter('kernel.bundles');
+        if (isset($bundles['HWIOAuthBundle'])) {
+            $config = $this->getResourceOwnersConfiguration($container);
+
+            $container->prependExtensionConfig('hwi_oauth', $config);
+        }
+    }
+
+    private function createConfigForResourceOwners($resourceOwners): array
+    {
+        $configForResourceOwners = [];
+        foreach ($resourceOwners as $resourceOwner => $resourceOwnerConfig) {
+            $options = json_decode($resourceOwnerConfig['options'] ?? '{}', true);
```

Comment:
This should be using the [Symfony OptionsResolver](https://symfony.com/doc/current/components/options_resolver.html), which can easily be used without being a service.

Reply:
Can you help me with this on monday